src/Logging/Logging.Batching/BatchLogService.cs

```diff
@@ -0,0 +1,87 @@
+﻿using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+namespace CodeMonkeys.Logging.Batching
+{
+    public abstract class BatchLogService<TOptions> : ScopedLogService<TOptions>
+        where TOptions : BatchLogOptions, new()
+    {
+        private BlockingCollection<LogMessage> _queue;
+
+        public override bool IsEnabled
+        {
+            get => base.IsEnabled;
+            protected set
+            {
+                if (value)
+                    Run();
+
+                base.IsEnabled = value;
+            }
+        }
+
+        protected BatchLogService(string context) 
+            : base(context)
+        {
+            Run();
+        }
+
+        protected abstract Task PublishMessageBatch(IEnumerable<LogMessage> messageBatch);
+
+        protected override void PublishMessage(LogMessage message)
+        {
+            if (_queue.IsAddingCompleted)
+                return;
+
+            try
+            {
+                _queue.Add(message);
+            }
+            catch { }
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Handle the exception or explain in a comment why it can be ignored.](https://app.codacy.com/gh/UltimateCodeMonkeys/CodeMonkeysMVVM/pullRequest?prid=5626772)